### ISSUE TITLE
Collections Enum.chunk_by

### ISSUE THREAD

**Issue by bigtunacan** [doomspork/elixir-school#478]
In the Collections chapter for Enum.chunk_by the following example is used.

    Enum.chunk_by(["one", "two", "three", "four", "five"], fn(x) -> String.length(x) end)

This results in 

[["one", "two"], ["three"], ["four", "five"]]

To a new comer (like myself) one might expect then that all strings of the same length would be grouped together if the function returns the same value, but this is not the case as can be seen with this example.

    Enum.chunk_by(["one", "two", "three", "four", "five", "six"], fn(x) -> String.length(x) end)

This results in 

[["one", "two"], ["three"], ["four", "five"], ["six"]]

**Comment by ybur-yug** [comment]
@bigtunacan How would you recommend clarifying this?

From the documentation for `Enum.chunk_by/2`:

```
                         def chunk_by(enumerable, fun)

Splits enumerable on every element for which fun returns a new value.

Returns a list of lists.

Examples

┃ iex> Enum.chunk_by([1, 2, 2, 3, 4, 4, 6, 7, 7], &(rem(&1, 2) == 1))
┃ [[1], [2, 2], [3], [4, 4, 6], [7, 7]]
```

So each time the function being applied has a different return, a new chunk is started.

Would it read more clearly as something like this?

```
If we need to group our collection based on something other than size, we can use the `chunk_by` method. It takes a given enumerable and a function, and when the return on that function changes a new group is started and begins the creation of the next:
```

Any suggestions? Happy to work on making this one more understandable.

CC @doomspork you have any thoughts on how it could be made more clear?

**Comment by doomspork** [comment]
This was resolved by @ybur-yug in #528!
